### PR TITLE
Add per-page theming option

### LINKED ISSUE #321: Per-page theming options
**Current situation**
Lesson pages currently all have the same color.

**Desired situation**
For each lessons pages we can specify what colors should be used via the lesson's `metadata` object.

### PR DESCRIPTION
**Pull request by shaavan**
resolves #321

- Add per page theming option.
- Removes layout.tsx for `HOMEADDRESS/en/chapters/chapter-<n>` since the link to the address results in a 404 page. And there is no page with `[slug]` and no `[lesson]`. Removing this file removes redundant complexity from the codebase.

Reference: https://savingsatoshi.com/en/chapters/chapter-1

### CODE REVIEW

**Comment by shaavan** [comment]
<h2>
Screenshot:
</h2>

Patch used to test the PR:

```diff
--- a/content/lessons/chapter-1/intro-1.tsx
+++ b/content/lessons/chapter-1/intro-1.tsx
@@ -4,6 +4,7 @@ import { Introduction, Text } from 'ui'
 export const metadata = {
   title: 'intro_one.title',
   image: '/assets/images/chapter-1-intro-1.jpg',
+  theme: 'bg-green'
 }
 
 export default function ReapingRewards({ lang }) {
```

|Before patch|After patch|
|------------|------------|
|image|image|

**Review by benalleng — CHANGES_REQUESTED** [review]
Code LGTM, Lets just add the relevant color themes from the following figma files
[intro-1](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=2374-25204&t=RwVyoIlNwUpAgbXD-0)
[intro-2](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=2374-25234&t=RwVyoIlNwUpAgbXD-0)
[genesis-4](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=2648-25633&t=RwVyoIlNwUpAgbXD-0)
[outro-1](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=2374-26622&t=RwVyoIlNwUpAgbXD-0)

**Comment by afterburn** [comment]
Code approved, feel free to merge after resolving @benalleng's comment.

**Comment by shaavan** [comment]
Updated from ec07a2f2a22e8d402fea6eaf1f792e54418b1b25 to 17555134f3c1e19d141d282d96fc3ba566f722ee ([pr325.01](https://github.com/shaavan/saving-satoshi/commits/pr325.01) -> [pr325.02](https://github.com/shaavan/saving-satoshi/commits/pr325.02), [diff](https://github.com/shaavan/saving-satoshi/compare/pr325.01..pr325.02))

Address @benalleng comment and made a few other changes.

**Changes:**

1. Updated code so the tooltip and dropdown menu follows the design theme.
2. Removed `use client` from all the lesson content files, as it was redundant and interfered with the proper reading of metadata content.
3. Updated lesson themes as per [Figma](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=2374-25204&t=RwVyoIlNwUpAgbXD-0) file.

**Comment by shaavan** [comment]
Updated from 17555134f3c1e19d141d282d96fc3ba566f722ee to e69d743c7cf8c7eed81c64c8c543b4f98bf31787 ([pr325.02](https://github.com/shaavan/saving-satoshi/commits/pr325.02) -> [pr325.03](https://github.com/shaavan/saving-satoshi/commits/pr325.03), [diff](https://github.com/shaavan/saving-satoshi/compare/pr325.02..pr325.03))
Addressed @benalleng suggestion

**Changes:**
1. Fixed error caused by making lesson render server-side.

**Comment by shaavan** [comment]
Updated from e69d743c7cf8c7eed81c64c8c543b4f98bf31787 to e5190765b6e0f7ca661cc957c134b2bbd1d299fe ([pr325.03](https://github.com/shaavan/saving-satoshi/commits/pr325.03) -> [pr325.04](https://github.com/shaavan/saving-satoshi/commits/pr325.04), [diff](https://github.com/shaavan/saving-satoshi/compare/pr325.03..pr325.04))

**Changes:**

- Introduced 'use client' back in input challenges to prevent the issue pointed out by @benalleng.

**Comment by shaavan** [comment]
@benalleng 

- Temporarily, I have introduced the 'use client' back in input challenges (genesis-2, transacting-2) because they did not have a specific theme for them, so we don't require to extract metadata in layout.tsx.
- For now, we can move forward with merging this PR, and solve this specific issue, in a followup.

**Comment by benalleng** [comment]
Lets go ahead and revert all client-side rendering changes and make a new PR exploring those changes as per our call

File: content/lessons/chapter-1/genesis-1.tsx [hunk 1]
```diff
@@ -1,4 +1,3 @@
-'use client'
 import { useTranslations } from 'hooks'
```
Comment (benalleng) [inline content/lessons/chapter-1/genesis-1.tsx:1]:
```suggestion
'use client'

import { useTranslations } from 'hooks'
```

File: content/lessons/chapter-1/genesis-3.tsx [hunk 2]
```diff
@@ -1,5 +1,3 @@
-'use client'
-
 import { useTranslations } from 'hooks'
```
Comment (benalleng) [inline content/lessons/chapter-1/genesis-3.tsx:1]:
```suggestion
'use client'

import { useTranslations } from 'hooks'
```

File: content/lessons/chapter-1/transacting-1.tsx [hunk 3]
```diff
@@ -1,5 +1,3 @@
-'use client'
-
 import { useTranslations } from 'hooks'
```
Comment (benalleng) [inline content/lessons/chapter-1/transacting-1.tsx:1]:
```suggestion
'use client'

import { useTranslations } from 'hooks'
```

File: content/lessons/chapter-1/transacting-3.tsx [hunk 4]
```diff
@@ -1,5 +1,3 @@
-'use client'
-
 import { useTranslations } from 'hooks'
```
Comment (benalleng) [inline content/lessons/chapter-1/transacting-3.tsx:1]:
```suggestion
'use client'

import { useTranslations } from 'hooks'
```

**Review by benalleng — CHANGES_REQUESTED** [review]
Lets go ahead and revert all client-side rendering changes that aren't absolutely necessary for the theme changes and make a new PR exploring those changes as per our call

**Comment by shaavan** [comment]
Updated from e5190765b6e0f7ca661cc957c134b2bbd1d299fe to 895cf4ec38730c26699aa697bed6182c73d66f51 ([pr325.04](https://github.com/shaavan/saving-satoshi/commits/pr325.04) -> [pr325.05](https://github.com/shaavan/saving-satoshi/commits/pr325.05), [diff](https://github.com/shaavan/saving-satoshi/compare/pr325.04..pr325.05))

Addressed @benalleng suggestion.

- Reverted changes to lesson files that are not directly relevant to this PR.
- The discussion around rendering lesson content on the server/client side shall be continued in a separate issue/PR.

File: content/lessons/chapter-1/transacting-1.tsx (hunk 3 above)
Comment (shaavan) [reply content/lessons/chapter-1/transacting-1.tsx]: Addressed in 895cf4ec38730c26699aa697bed6182c73d66f51

File: content/lessons/chapter-1/genesis-3.tsx (hunk 2 above)
Comment (shaavan) [reply content/lessons/chapter-1/genesis-3.tsx]: Addressed in 895cf4ec38730c26699aa697bed6182c73d66f51
Thanks!

**Review by benalleng — APPROVED** [review]

**Comment by ecurrencyhodler** [comment]
I'm unable to merge as the following conflicting files must be addressed:

- Conflicting files
- components/Navbar/NavbarDesktop/Tab.tsx
- content/lessons/chapter-1/genesis-4.tsx
- content/lessons/chapter-1/intro-2.tsx
- content/lessons/chapter-1/outro-1.tsx

**Comment by shaavan** [comment]
Updated from 895cf4ec38730c26699aa697bed6182c73d66f51 to e41911a61663ae2d2940f344c864050f0529e627 ([pr325.05](https://github.com/shaavan/saving-satoshi/commits/pr325.05) -> [pr325.06](https://github.com/shaavan/saving-satoshi/commits/pr325.06))
Addressed @ecurrencyhodler comments

- Rebased over master and resolved merge conflicts.
- Made changes to render layout.tsx on the client side. This solves the issue of needing to render lessons on the server side.

**Pull request merged**